File: odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/OkPostingActivity.java
package ru.ok.android.sdk;

import org.json.JSONException;
import org.json.JSONObject;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.webkit.WebView;

public class OkPostingActivity extends AbstractWidgetActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.ok_posting_activity);
        prepareWebView();

        Bundle bundle = getIntent().getExtras();
        if (bundle != null) {
            args.put("st.attachment", bundle.getString(Shared.PARAM_ATTACHMENT));
            args.put("st.utext", bundle.getBoolean(Shared.PARAM_USER_TEXT_ENABLE, false) ? "on" : "off");
        }

        String url = prepareUrl(null);
        ((WebView) findViewById(R.id.web_view)).loadUrl(url);
    }

    private void prepareWebView() {
        WebView webView = (WebView) findViewById(R.id.web_view);
        webView.setWebViewClient(new OkWidgetViewClient(this));
        webView.getSettings().setJavaScriptEnabled(true);
    }

    @Override
    protected int getCancelledMessageId() {
        return R.string.posting_canceled;
    }

    @Override
    protected String getWidgetId() {
        return "WidgetMediatopicPost";
    }

    @Override
    protected void processResult(String result) {
        Odnoklassniki odnoklassniki = Odnoklassniki.getInstance();
        Intent resultIntent = new Intent();
        if (odnoklassniki != null) {
            try {
                JSONObject json = new JSONObject(result);
                String type = json.getString(Shared.PARAM_TYPE);
                if ("error".equals(type)) {
                    resultIntent.putExtra(Shared.PARAM_ERROR, json.getString(Shared.PARAM_MESSAGE));
                } else {
                    resultIntent.putExtra(Shared.PARAM_RESULT, json.toString());
                }
            } catch (JSONException e) {
                resultIntent.putExtra(Shared.PARAM_ERROR, result);
            }
        }
        setResult(Activity.RESULT_OK, resultIntent);
        finish();
    }
}

File: odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/OkPostingActivity.kt
package ru.ok.android.sdk

import android.annotation.SuppressLint
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.webkit.WebView
import org.json.JSONException
import org.json.JSONObject

internal class OkPostingActivity : AbstractWidgetActivity() {

    override val cancelledMessageId: Int
        get() = R.string.posting_canceled

    override val widgetId: String
        get() = "WidgetMediatopicPost"

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        prepareWebView()

        val bundle = intent.extras
        if (bundle != null) {
            args["st.attachment"] = bundle.getString(PARAM_ATTACHMENT) ?: "{}"
            args["st.utext"] = if (bundle.getBoolean(PARAM_USER_TEXT_ENABLE, false)) "on" else "off"
        }

        val url = prepareUrl(null)
        findViewById<WebView>(R.id.web_view).loadUrl(url)
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun prepareWebView() {
        val webView = findViewById<WebView>(R.id.web_view)
        webView.webViewClient = OkWidgetViewClient(this)
        webView.settings.javaScriptEnabled = true
    }

    override fun processResult(result: String?) {
        val resultIntent = Intent()
        try {
            val json = JSONObject(result)
            val type = json.getString(PARAM_TYPE)
            if ("error" == type) {
                resultIntent.putExtra(PARAM_ERROR, json.getString(PARAM_MESSAGE))
            } else {
                resultIntent.putExtra(PARAM_RESULT, json.toString())
            }
        } catch (e: JSONException) {
            resultIntent.putExtra(PARAM_ERROR, result)
        }
        setResult(Activity.RESULT_OK, resultIntent)
        finish()
    }
}
